backstory/src/test/java/com/google/sps/story/DatamuseRequestClientTest.java

```diff
@@ -0,0 +1,337 @@
+import com.google.sps.story.data.HttpUrlStreamHandler;
+import com.google.sps.story.data.DatamuseRequestClient;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.net.MalformedURLException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLStreamHandler;
+import java.net.URLStreamHandlerFactory;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.BDDMockito;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+/** Quality tests for DatamuseRequestClient */
+@RunWith(JUnit4.class)
+public final class DatamuseRequestClientTest {
+
+  /** a default url to test with */
+  private static final String DEFAULT_URL = "http://sample-website.com/fake?";
+
+  /** the stream handlet to use for tests in this class (necessary to mock URLConnection) */
+  private static HttpUrlStreamHandler httpUrlStreamHandler;
+ 
+  /** 
+   * Set up the URLStreamHandlerFactory with the HTTPUrlStreamHandler 
+   * in order to mock URLConnections to properly test methods 
+   * that call outside URLs. Code comes from this tutorial: 
+   * https://claritysoftware.co.uk/mocking-javas-url-with-mockito/.
+   */
+  @BeforeClass
+  public static void setUpURLStreamHandlerFactory() {
+    // Allows for mocking URL connections
+    URLStreamHandlerFactory urlStreamHandlerFactory = mock(URLStreamHandlerFactory.class);
+    URL.setURLStreamHandlerFactory(urlStreamHandlerFactory);
+ 
+    httpUrlStreamHandler = new HttpUrlStreamHandler();
+    BDDMockito.given(urlStreamHandlerFactory.createURLStreamHandler("http")).willReturn(httpUrlStreamHandler);
+  }
+
+  /**
+   * Reset the URLConnections of this stream handler before every method,
+   * so that URLConnections injected for other tests do not mess 
+   * with the current test. Code taken from this tutorial: 
+   * https://claritysoftware.co.uk/mocking-javas-url-with-mockito/.
+   */
+  @Before
+  public void reset() {
+    httpUrlStreamHandler.resetConnections();
+  }
+
+  /** 
+   * Check that IllegalArgumentException thrown for a multiple
+   * word noun.
+   */
+  @Test
+  public void multipleWordNoun() throws Exception {
+    final String MULTIWORD_NOUN = "multiword noun";
+    final int DEFAULT_MAX = 5;
+
+    DatamuseRequestClient client = new DatamuseRequestClient(DEFAULT_URL);
+
+    try {
+      client.fetchRelatedAdjectives(MULTIWORD_NOUN, DEFAULT_MAX);
+    } catch (IllegalArgumentException exception) {
+      return; // test should pass if this exception is thrown
```

Comment:
Can you use the `expected` annotation instead (doesn't marking this method with `throws` make this possible)?

Question also applies to tests below.

Reply:
I think I can now. I'll make the change.

Reply:
Done